x-pack/plugins/synthetics/public/apps/synthetics/components/common/alerting_callout/alerting_callout.tsx

```diff
@@ -30,45 +43,74 @@ export const AlertingCallout = ({ isAlertingEnabled }: { isAlertingEnabled?: boo
   const syntheticsLocators = useSyntheticsStartPlugins()?.share?.url.locators;
   const locator = syntheticsLocators?.get(syntheticsSettingsLocatorID);
 
-  useEffect(() => {
-    async function generateUrl() {
-      const settingsUrl = await locator?.getUrl({});
-      setUrl(settingsUrl);
-    }
-    generateUrl();
+  const { data: url } = useFetcher(() => {
+    return locator?.getUrl({});
   }, [locator]);
 
-  const hasDefaultConnector =
-    Boolean(defaultConnectors?.length) || (!settingsLoading && Boolean(defaultConnectors?.length));
   const hasAlertingConfigured =
     isAlertingEnabled ??
     (monitorsLoaded &&
       monitors.some((monitor) => monitor[ConfigKey.ALERT_CONFIG]?.status?.enabled));
 
   const showCallout = url && !hasDefaultConnector && hasAlertingConfigured;
+  const missingRules = !rulesLoading && Object.keys(defaultRules ?? {}).length !== 2 && !canSave;
```

Comment:
`Object.keys().length` condition here is fragile and should be improved.

For a read only user, it contains a third property, resulting in both of the callouts always shown despite being already configured (tested on a non-default space).

<img width="1568" alt="Screenshot 2023-07-11 at 15 53 51" src="https://github.com/elastic/kibana/assets/2748376/1b32eb6b-85b3-4686-8695-1ab24418b8d1">